src/algebra/classical_lie_algebras.lean

```diff
@@ -90,4 +130,199 @@ end
+
+/-- The symplectic Lie algebra: skew-adjoint matrices with respect to the canonical skew-symmetric
+bilinear form. -/
+def sp : lie_subalgebra R (matrix (l ⊕ l) (l ⊕ l) R) :=
+  skew_adjoint_matrices_lie_subalgebra (J l R)
+
+end symplectic
+
+namespace orthogonal
+
+/-- The definite orthogonal Lie subalgebra: skew-adjoint matrices with respect to the symmetric
+bilinear form defined by the identity matrix. -/
+def so : lie_subalgebra R (matrix n n R) :=
+  skew_adjoint_matrices_lie_subalgebra (1 : matrix n n R)
+
+/-- The indefinite diagonal matrix with `p` 1s and `q` -1s. -/
+def indefinite_diagonal : matrix (p ⊕ q) (p ⊕ q) R :=
+  matrix.diagonal $ sum.elim (λ _, 1) (λ _, -1)
+
+/-- The indefinite orthogonal Lie subalgebra: skew-adjoint matrices with respect to the symmetric
+bilinear form defined by the indefinite diagonal matrix. -/
+def so' : lie_subalgebra R (matrix (p ⊕ q) (p ⊕ q) R) :=
+  skew_adjoint_matrices_lie_subalgebra $ indefinite_diagonal p q R
+
+/-- A matrix for transforming the indefinite diagonal bilinear form into the definite one, provided
+the parameter `i` is a square root of -1. -/
+def Pso (i : R) : matrix (p ⊕ q) (p ⊕ q) R :=
+  matrix.diagonal $ sum.elim (λ _, 1) (λ _, i)
+
+lemma Pso_inv {i : R} (hi : i*i = -1) : (Pso p q R i) * (Pso p q R (-i)) = 1 :=
+begin
+  ext x y, rcases x; rcases y,
+  { -- x y : p
+    by_cases h : x = y; simp [Pso, indefinite_diagonal, h], },
+  { -- x : p, y : q
+    simp [Pso, indefinite_diagonal], },
+  { -- x : q, y : p
+    simp [Pso, indefinite_diagonal], },
+  { -- x y : q
+    by_cases h : x = y; simp [Pso, indefinite_diagonal, h, hi], },
+end
+
+lemma is_unit_Pso {i : R} (hi : i*i = -1) : is_unit (Pso p q R i) :=
+⟨{ val     := Pso p q R i,
+   inv     := Pso p q R (-i),
+   val_inv := Pso_inv p q R hi,
+   inv_val := by { apply matrix.nonsing_inv_left_right, exact Pso_inv p q R hi, }, },
+rfl⟩
+
+lemma indefinite_diagonal_transform {i : R} (hi : i*i = -1) :
+  (Pso p q R i)ᵀ ⬝ (indefinite_diagonal p q R) ⬝ (Pso p q R i) = 1 :=
+begin
+  ext x y, rcases x; rcases y,
+  { -- x y : p
+    by_cases h : x = y; simp [Pso, indefinite_diagonal, h], },
+  { -- x : p, y : q
+    simp [Pso, indefinite_diagonal], },
+  { -- x : q, y : p
+    simp [Pso, indefinite_diagonal], },
+  { -- x y : q
+    by_cases h : x = y; simp [Pso, indefinite_diagonal, h, hi], },
+end
+
+/-- An equivalence between the indefinite and definite orthogonal Lie algebras, over a ring
+containing a square root of -1. -/
+noncomputable def so_indefinite_equiv {i : R} (hi : i*i = -1) : so' p q R ≃ₗ⁅R⁆ so (p ⊕ q) R :=
+begin
+  apply (skew_adjoint_matrices_lie_subalgebra_equiv
+    (indefinite_diagonal p q R) (Pso p q R i) (is_unit_Pso p q R hi)).trans,
+  apply lie_algebra.equiv.of_eq,
+  ext A, rw indefinite_diagonal_transform p q R hi, refl,
+end
+
+lemma so_indefinite_equiv_apply {i : R} (hi : i*i = -1) (A : so' p q R) :
+  ↑(so_indefinite_equiv p q R hi A) = (Pso p q R i)⁻¹ ⬝ ↑A ⬝ (Pso p q R i) :=
```

Comment:
Could you replace those up-arrows by type ascription? This is easier to read.

Reply:
I certainly can but are you sure you prefer how it looks. It will become:
```lean
(so_indefinite_equiv p q R hi A : matrix (p ⊕ q) (p ⊕ q) R) = (Pso p q R i)⁻¹ ⬝ (A : matrix (p ⊕ q) (p ⊕ q) R) ⬝ (Pso p q R i)
```

Reply:
On second thought, I'll push these changes since you clearly asked for them.